pkg/leaderelection/leader_election.go

```diff
@@ -41,7 +41,7 @@ func NewResourceLock(corev1 v1.CoreV1Interface, coordinationV1 v12.CoordinationV
 	}
 
 	// Leader id, needs to be unique
-	return resourcelock.New(resourcelock.ConfigMapsResourceLock,
+	return resourcelock.New(resourcelock.ConfigMapsLeasesResourceLock,
```

Comment:
I'm not entirely sure, but this change seems to require additional permissions:

```yaml
 apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs: ...
```
If so, this is a breaking change unfortunately.

Reply:
In fact, this caused a failed deployment for us because we did not have this RBAC permission in place when upgrading to 1.2.0.

Reply:
@honnix so sorry about not exposing this in the release notes, thanks for tracking it down - it will be added. Are these additional RBAC permissions an issue?

It looks like the `ConfigMapLeasesResourceLock` has been [the default ](https://github.com/kubernetes-sigs/controller-runtime/blob/1730628f118be1128787012c164379e6ac46c75f/pkg/manager/manager.go#L143-L156 ) for a very long time and we should have migrated over sooner. Now support for `ConfigMapResourceLock` has [been removed](https://github.com/kubernetes/client-go/commit/276ea3ed979947d7cdd4b3d708862245ddcd8883).

Reply:
@hamersaw No worries. Just wanted to give you folks a heads-up. It was fairly easy to solve on our side and it is all good now.